packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -38,10 +36,32 @@ export class FabricBase extends React.Component<
     if (win) {
       this._disposables.push(on(win, 'mousedown', this._onMouseDown, true), on(win, 'keydown', this._onKeyDown, true));
     }
+
+    const classNames = this._getClassNamesHelper();
+    if (this.props.applyThemeToBody) {
+      const currentDoc: Document = getDocument(this._rootElement.current)!;
+      currentDoc.body.classList.add(classNames.bodyThemed);
+    }
   }
 
   public componentWillUnmount(): void {
     this._disposables.forEach((dispose: () => void) => dispose());
+
```

Comment:
There could be some weird edge cases here that are unlikely, but should guard against. If a developer renders this control with `applyThemeToBody=true`, then sets `applyThemeToBody=false`, then unmounts, this code will not have the intended effect.

Instead, what about having `componentDidMount()` define a callback to clean up (much like `_disposables`)? Has the added benefit of evaluating `_getClassNamesHelper()` fewer times.

Reply:
You'll probably want to clean-up regardless of props. Imagine the scenario where you render this on a page using the new props to add a class to the `document.body`. Then you remove that component while performing an in-place navigation (not a hard refresh of the document). That class will remain on `document.body` and impact other components until the document is hard-refreshed. I'm thinking of SPFx WebParts in particular.

Reply:
I'm slightly confused here - so is it recommended to do a `this._disposables.forEach((dispose: () => void) => dispose());` in `componentDidMount()` too when it's already happening in `componentDidUnmount`? And is it better to do it even if `props.applyThemeToBody` is passed in? To be more clear, what kind of an action would cause there to be a component to be removed during an in-place navigation? I'm trying to think of an example - like while scrolling or navigating to the last page, somehow the component this prop is on is removed?

Reply:
I was advocating for removing the classes previously applied by this component when the component is unmounted via `componentDidUnmount`. This way the component cleans-up after itself and doesn't leave artifacts like classNames on `document.body`. That can have side-effects when navigating in an app where we may want the behavior on some pages but not others.

Reply:
Done.